### DISCUSSION TITLE
AR is not working on android devices on page load

### DISCUSSION THREAD

**Discussion by paavamaani** [google/model-viewer#2440] [Q&A]
**Description**

Hi Team,

We are using model viewer library to display 3d model in our website. Where we have a specific use case don’t work.

Sample URL: http://www.domain.com/path?launchAr=modelId Won’t be able to provide actual URL as it is not my own project. But an e-commerce site.

We are using a custom parameter `launchAr` to target specific model(product) which need to show AR view. We are trying to listen to an event `model-visibility` on page load and trigger activateAR() for specific model on page. But, the model viewer return `ar-status` as `failed` even though device support AR and this issue occurs only for android devices it works fine in iOS

But, if we have any other model(product) on same page and we try to click a button/link explicitly to launch AR. AR view seems to work fine in iOS and android.

Basically activateAR() is not working on page load for android devices. It works fine if I try to click/button or link explicitly

We are initially using older version of model viewer. But, we upgraded it to latest 1.7.1 and still we the issue persist

Could you please help us with this

**model-viewer**: v1.7.1

**Browser Affected**

- [x] Chrome, version: 90.0.4430.210
- [ ] Edge
- [ ] Firefox
- [ ] IE
- [ ] Safari

**OS**

- [x] Android, version: 11
- [ ] iOS
- [ ] Linux
- [ ] MacOS
- [ ] Windows

**AR**

- [ ] WebXR
- [x] SceneViewer
- [ ] QuickLook

**Comment by elalish** [comment]
If you check the console, you will likely find an error that says something like "WebXR requires user activation". This is a Chrome security requirement, and the reason we have a button to launch AR.

**Reply by paavamaani** [reply]
Hi elalish,

Thank you for your quick feedback

I have one question regarding user interaction.

Is the user interaction we require for trigger AR a message prompting for permission to open the camera or is this something the user interaction on browser?

Regards,
Paavamaani

**Reply by elalish** [reply]
The browser takes care of prompting the user for any required permissions once the click happens.

**Reply by paavamaani** [reply]
We have custom button on click we call activareAR() in this case AR is launched.

So any user interaction on page not only the default button to launch ar but also custom button is consider as user interaction and it works?

**Reply by elalish** [reply]
Yes, it should. Just make sure there aren't any delays or Chrome will consider it stale.

**Reply by ghost** [reply]
Hi @elalish,
So, for example, if we take the last **real** user action (tap the screen, press any button, drag/scroll, etc.), there is a time-lapse limit to trigger the AR view automatically, because if it exceeds that time, then Google Chrome considers the user is idle and it does not allow to trigger the AR view.

Is it like this? Do you know how much time-lapse are we talking about?
Thanks a lot.

**Reply by elalish** [reply]
Not sure, but it's short. Basically you should just put the activation in a small click handler.
